src/main/java/org/opentripplanner/middleware/utils/ConfigUtils.java

```diff
@@ -53,6 +58,38 @@ public static void loadConfig(String[] args) throws IOException {
             envConfigStream = new FileInputStream(new File(args[0]));
         }
         envConfig = yamlMapper.readTree(envConfigStream);
+        validateConfig();
+    }
+
+    /**
+     * Validate the environment configuration against the environment configuration schema. If the config is not available,
+     * does not match the schema or an exception is thrown, exit the application.
+     */
+    private static void validateConfig() {
+        if ("false".equals(getConfigPropertyAsText("VALIDATE_ENVIRONMENT_CONFIG", "true"))) {
+            LOG.warn("Environment configuration schema validation disabled.");
+            return;
+        }
+        try {
+            if (envConfig == null) {
+                throw new IllegalArgumentException("Environment configuration not available to validate!");
+            }
+            FileInputStream envSchemaStream = new FileInputStream(new File(DEFAULT_ENV_SCHEMA));
+            JsonNode envSchema = yamlMapper.readTree(envSchemaStream);
+            if (envSchema == null) {
+                throw new IllegalArgumentException("Environment configuration schema not available.");
+            }
+            ProcessingReport report = JsonSchemaFactory
+                    .byDefault()
+                    .getValidator()
+                    .validate(envSchema, envConfig);
+            if (!report.isSuccess()) {
+                throw new IllegalArgumentException(report.toString());
+            }
+        } catch (IOException | IllegalArgumentException | ProcessingException e) {
+            LOG.error("Unable to validate environment configuration.", e);
```

Comment:
Here and elsewhere in the log statements, I think we should be more explicit about what we're validating, just so that a user of the software isn't scratching their head about the somewhat vague reference to "environment". In other words, we should just say `env.yml` instead of `environment configuration`. For example: "Unable to validate config file (env.yml)". (Similar to how we refer to it in the loadConfig method)

Reply:
Updated.